src/api/entities/Account.ts

```diff
@@ -0,0 +1,206 @@
+import { Context, PolymeshError } from '~/base';
+import { transactions } from '~/middleware/queries';
+import { Query, TransactionOrderByInput } from '~/middleware/types';
+import {
+  AccountBalance,
+  Ensured,
+  ErrorCode,
+  ExtrinsicData,
+  ResultSet,
+  SubCallback,
+  UnsubCallback,
+} from '~/types';
+import {
+  addressToKey,
+  calculateNextKey,
+  extrinsicIdentifierToTxTag,
+  identityIdToString,
+  stringToAccountId,
+  txTagToExtrinsicIdentifier,
+} from '~/utils';
+
+export interface UniqueIdentifiers {
+  address: string;
+}
+
+/**
+ * Represents an account in the Polymesh blockchain. Accounts can hold POLYX, control Identities and vote on proposals (among other things)
+ */
+export class Account extends Entity<UniqueIdentifiers> {
+  /**
+   * @hidden
+   * Check if a value is of type [[UniqueIdentifiers]]
+   */
+  public static isUniqueIdentifiers(identifier: unknown): identifier is UniqueIdentifiers {
+    const { address } = identifier as UniqueIdentifiers;
+
+    return typeof address === 'string';
+  }
+
+  /**
+   * Polymesh-specific address of the account. Serves as an identifier
+   */
+  public address: string;
+
+  /**
+   * public key of the account. This is a hex representation of the address that is transversal to any Substrate chain
+   */
+  public key: string;
+
+  /**
+   * @hidden
+   */
+  public constructor(identifiers: UniqueIdentifiers, context: Context) {
+    super(identifiers, context);
+
+    const { address } = identifiers;
+
+    this.address = address;
+    this.key = addressToKey(address);
+  }
+
+  /**
+   * Get the free/locked POLYX balance of the account
+   *
+   * @note can be subscribed to
+   */
+  public getBalance(): Promise<AccountBalance>;
+  public getBalance(callback: SubCallback<AccountBalance>): Promise<UnsubCallback>;
+
+  // eslint-disable-next-line require-jsdoc
+  public getBalance(
+    callback?: SubCallback<AccountBalance>
+  ): Promise<AccountBalance | UnsubCallback> {
+    const { context, address } = this;
+
+    if (callback) {
+      return context.accountBalance(address, callback);
+    }
+
+    return context.accountBalance(address);
+  }
+
+  /**
+   * Retrieve the Identity associated to this Account
+   *
+   * @throws if there is no identity associated to the account
+   */
+  public async getIdentity(): Promise<Identity> {
+    const {
+      context: {
+        polymeshApi: {
+          query: { identity },
+        },
+      },
+      context,
+      address,
+    } = this;
+
+    try {
+      const identityIdWrapper = await identity.keyToIdentityIds(
+        stringToAccountId(address, context)
+      );
+      const did = identityIdToString(identityIdWrapper.unwrap().asUnique);
+
+      return new Identity({ did }, context);
+    } catch (err) {
+      throw new PolymeshError({
+        code: ErrorCode.IdentityNotPresent,
+        message: 'The current account does not have an associated identity',
+      });
```

Comment:
Should it throw? Maybe it should return null. Will it impact on how we are catching this problem where Identity is required?